File: backend/src/main/java/vn/edu/ut/pbms/service/VehicleTypeService.java
package vn.edu.ut.pbms.service;

import vn.edu.ut.pbms.dto.request.VehicleTypeRequestDTO;
import vn.edu.ut.pbms.dto.response.VehicleTypeResponseDTO;

import java.util.List;

/**
 * Service interface for VehicleType business logic.
 * Defines all operations for managing vehicle types.
 */
public interface VehicleTypeService {

    /**
     * Retrieve all vehicle types.
     *
     * @return list of all vehicle types as response DTOs
     */
    List<VehicleTypeResponseDTO> getAllVehicleTypes();

    /**
     * Create a new vehicle type.
     *
     * @param requestDTO the vehicle type data from the client
     * @return the created vehicle type as a response DTO
     */
    VehicleTypeResponseDTO createVehicleType(VehicleTypeRequestDTO requestDTO);

    /**
     * Update an existing vehicle type.
     *
     * @param id         the ID of the vehicle type to update
     * @param requestDTO the updated vehicle type data
     * @return the updated vehicle type as a response DTO
     */
    VehicleTypeResponseDTO updateVehicleType(Long id, VehicleTypeRequestDTO requestDTO);

    /**
     * Deactivate a vehicle type by setting its status to INACTIVE.
     *
     * @param id the ID of the vehicle type to deactivate
     * @return the deactivated vehicle type as a response DTO
     */
    VehicleTypeResponseDTO deactivateVehicleType(Long id);
}

File: frontend/src/services/vehicle-type.service.ts
import axiosClient from "@/lib/axios-client";
import { VehicleType, VehicleTypeRequest } from "@/features/vehicle-types/types/vehicle-type.type";

const BASE_PATH = "/vehicle-types";

export const getVehicleTypes = async (): Promise<VehicleType[]> => {
  const response = await axiosClient.get<VehicleType[]>(BASE_PATH);
  return response.data;
};

export const createVehicleType = async (data: VehicleTypeRequest): Promise<VehicleType> => {
  const response = await axiosClient.post<VehicleType>(BASE_PATH, data);
  return response.data;
};

export const updateVehicleType = async (
  id: number,
  data: VehicleTypeRequest
): Promise<VehicleType> => {
  const response = await axiosClient.put<VehicleType>(`${BASE_PATH}/${id}`, data);
  return response.data;
};

export const deactivateVehicleType = async (id: number): Promise<VehicleType> => {
  const response = await axiosClient.patch<VehicleType>(`${BASE_PATH}/${id}/deactivate`);
  return response.data;
};
